src/org/opensolaris/opengrok/history/Repository.java

```diff
@@ -388,7 +392,69 @@ boolean supportsSubRepositories() {
     }
 
     public DateFormat getDateFormat() {
-        return new SimpleDateFormat(datePattern, Locale.US);
+        return new DateFormat() {
+            private final SimpleDateFormat[] formatters = new SimpleDateFormat[datePatterns.length];
+
+            {
+                // initialize date formatters
+                for (int i = 0; i < datePatterns.length; i++) {
+                    formatters[i] = new SimpleDateFormat(datePatterns[i], Locale.US);
```

Comment:
I'm not quite sure here about the locale. Maybe the default would be better choice.